### PR TITLE
(chore): update developer documentation

### PR DESCRIPTION
**Pull request by Jipperism**
- Updates developer documentation for contributing to the sdk.
- Updates developer documentation for consuming the SDK.
- Removes redundant references to env vars.
- Removes redundant references to outdated package manager yarn.
- Updates typedoc version and configuration to include handwritten guides in the generated docs.
- Removes outdated github templates.
- Updates JSDoc annotations.

### CODE REVIEW

File: .vscode/settings.json
Comment (bitbeckers) [inline .vscode/settings.json:1]: No .vscode settings in the repos please

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -22,11 +22,11 @@ Before you submit a PR, make sure that:
 - You code doesn't raise any ESlint warning
 - Your changes are explained in your PR
 
-When in doubt, [Create a new issue](https://github.com/LooksRare/sdk-v2/issues/new) to discuss your proposal first.
+When in doubt, [Create a new issue](https://github.com/hypercerts-org/marketplace-sdk/issues/new) to discuss your proposal first.
 
 ### Release
 
 - Create a [personal access token](https://github.com/settings/tokens/new?scopes=repo&description=release-it) (Don't change the default scope)
 - Create an `.env` (copy `.env.template`) and set you github personal access token.
```
Comment (bitbeckers) [inline CONTRIBUTING.md:30]: .env.example was removed, can't find .env.template?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -22,11 +22,11 @@ Before you submit a PR, make sure that:
 - You code doesn't raise any ESlint warning
 - Your changes are explained in your PR
 
-When in doubt, [Create a new issue](https://github.com/LooksRare/sdk-v2/issues/new) to discuss your proposal first.
+When in doubt, [Create a new issue](https://github.com/hypercerts-org/marketplace-sdk/issues/new) to discuss your proposal first.
 
 ### Release
 
 - Create a [personal access token](https://github.com/settings/tokens/new?scopes=repo&description=release-it) (Don't change the default scope)
```
Comment (bitbeckers) [inline CONTRIBUTING.md:29]: Are we using release-it?

File: doc/_media/cancelOrders.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+# How to cancel orders
+
+## Cancel orders using order nonces
+
+This method is used to invalidate the `orderNonce`. If multiple maker orders share the same nonce, they will all be cancelled and non-executable. An `orderNonce` is also invalidated once an order with that nonce is executed.
+
+```ts
+import { HypercertExchangeClient, ChainId } from "@hypercerts-org/marketplace-sdk";
+
+const hypercertExchangeClient = new HypercertExchangeClient(ChainId.MAINNET, provider, signer);
```
Comment (bitbeckers) [inline doc/_media/cancelOrders.md:12]: We're not on MAINNET.

**Review by bitbeckers — CHANGES_REQUESTED** [review]

File: .vscode/settings.json
Comment (Jipperism) [reply .vscode/settings.json:1]: Removed them.

File: doc/_media/cancelOrders.md (hunk 3 above)
Comment (Jipperism) [reply doc/_media/cancelOrders.md]: Changed it to OPTIMISM.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Jipperism) [reply CONTRIBUTING.md]: Removed the entire file as it's not up to date with our contribution guidelines.

**Review by pheuberger — COMMENTED** [review]
Looks like you're not only removing old code, but also bumping the version number and changing some magic number. I would advocate for even more granular commits, because I don't know why you're doing any of that (except the version bump).

File: src/HypercertExchangeClient.ts [hunk 4]
```diff
@@ -351,7 +319,7 @@ export class HypercertExchangeClient {
    * @param maker Maker order
    * @param taker Taker order
    * @param signature Signature of the maker order
-   * @param merkleTree If the maker has been signed with a merkle tree
+   * @param merkleTree Optional merkle tree
```
Comment (pheuberger) [inline src/HypercertExchangeClient.ts:322]: Why would the user use a Merkle tree? I think that would be a much better parameter description.

File: src/HypercertExchangeClient.ts [hunk 5]
```diff
@@ -766,7 +708,7 @@ export class HypercertExchangeClient {
   }
 
   /**
-   * Register the order with hypercerts marketplace API.
+   * Register the order with hypercerts marketplace API
```
Comment (pheuberger) [inline src/HypercertExchangeClient.ts:711]: ... with the Hypercerts ...

File: src/utils/api.ts [hunk 6]
```diff
@@ -124,9 +126,10 @@ export class ApiClient {
     }
 
     return await getOrders({ signer, chainId: chainId ? BigInt(chainId) : undefined }, this._urqlClient);
-  };
+  }
 
   /**
+   * @deprecated use GraphQL api instead instead
```
Comment (pheuberger) [inline src/utils/api.ts:132]: Same as above.

File: typedoc.json [hunk 7]
```diff
@@ -1,13 +1,12 @@
 {
   "$schema": "https://typedoc.org/schema.json",
-  "entryPoints": ["./src/HypercertExchangeClient.ts", "./src/types.ts"],
+  "entryPoints": ["./src/HypercertExchangeClient.ts", "./src/types.ts", "./guides/readme.md"],
```
Comment (pheuberger) [inline typedoc.json:3]: Isn't it `README.md`?

**Review by pheuberger — COMMENTED** [review]
Great stuff. Thank you so much for making this! It's mostly just nits and a couple of clarifying questions from my side. Being new to the project, I have big gaps in my knowledge and understanding which shows in my questions. Feel free to not elaborate on some of them if you think they're just too preposterous to cover hehe.

Please also don't be put off by the amount of comments. I was very thorough and highlighted very minor things. Generally this is great!

File: README.md [hunk 8]
```diff
@@ -1,101 +1,138 @@
...
+1. User A creates maker order and signs it
+1. User A registers maker order with API
+1. Signature on maker order gets verified
+1. Order gets stored in data postgres DB
+1. Order will live in DB until deleted
+1. Order will be visible to other users as long as it's valid.
+1. An order being executed or canceled (or many other reasons) might render it being invalid.
+    1. User B fetches order from API
+    1. User B creates taker order for maker order
+    1. User B signs taker order against maker order using the `HypercertExchangeClient`
```
[the remark this answers is not in the document]
Comment (Jipperism) [reply README.md:55]: Yes.

File: README.md [hunk 9]
```diff
@@ -1,101 +1,138 @@
...
+    1. User B calls the `executeOrder` method on the `HypercertExchangeClient` with the taker order, which calls the `HypercertExchange` contract in turn.
+    1. The `HypercertExchange` contract verifies the signature and executes the order
```
[the remark this answers is not in the document]
Comment (Jipperism) [reply README.md:57]: Yep, it is.

[the remark this answers is not in the document]
Comment (Jipperism) [reply README.md:57]: Is that something that should be added?

File: README.md [hunk 10]
```diff
@@ -1,101 +1,138 @@
...
+    1. The `HypercertExchange` contract will emit an event that the order has been executed, which is picked up by the indexer which revalidates the order and updates the errors codes and validity in the DB.
+1. Once an maker order is invalidated it's only visible to User A.
+1. Maker order can be deleted or permanently rendered invalid by declaring the nonce invalid by User A at any time.
+
+> [!WARNING]
+>
+> The indexer is not listing to the OrderExecuted event yet, so the order will not be updated in the DB, unless reverified manually.
```
[the remark this answers is not in the document]
Comment (Jipperism) [reply README.md:64]: Correct.

File: guides/cancelOrders.md [hunk 11]
```diff
@@ -7,40 +14,16 @@
 This method is used to invalidate the `orderNonce`. If multiple maker orders share the same nonce, they will all be cancelled and non-executable. An `orderNonce` is also invalidated once an order with that nonce is executed.
 
 ```ts
-import { LooksRare, ChainId } from "@looksrare/sdk-v2";
```
[the remark this answers is not in the document]
Comment (Jipperism) [reply guides/cancelOrders.md:10]: Yep. We renamed the main export from `LooksRare` to `HypercertExchangeClient`. It's been like that since the start, but we didn't update the docs yet to reflect that.

File: guides/createDirectFractionsSaleMakerAsk.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+// The ID of the fraction that will be put on sale
+const fractionId = 13601086205829910384631083059047775411896320n;
+
+// Create the fractional sale order for a hypercert
+// This will also take care of fetching and setting the order nonce
+const order = await hypercertExchangeClient.createDirectFractionsSaleMakerAsk({
+  startTime: Date.now() / 1000, // Use it to create an order that will be valid in the future (Optional, Default to now)
+  endTime: Date.now() / 1000 + 60 * 60, // If you use a timestamp in ms, the function will revert (this order will be valid for one hour)
+  itemIds: [fractionId], // ID of the fraction that will be put on sale
+  price: parseUnits("1", 18), // Price for the entire fraction, in this example we're selling the entire fraction for 1 USDC.
```
[the remark this answers is not in the document]
Comment (Jipperism) [reply guides/createDirectFractionsSaleMakerAsk.md:38]: It refers to a fraction, with is a part of a hypercert. I'm not really sure how to rephrase this to be honest.

[the remark this answers is not in the document]
Comment (Jipperism) [reply guides/createDirectFractionsSaleMakerAsk.md:38]: Gave it a shot, let me know what you think.

File: guides/createFractionalSaleMakerAsk.md [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+```ts
+import { parseEther } from "ethers";
+import { HypercertExchangeClient, ChainId, CollectionType, StrategyType } from "@hypercerts-org/marketplace-sdk";
+
+const hypercertExchangeClient = new HypercertExchangeClient(ChainId.OPTIMISM, provider, signer);
+
+// The ID of the fraction that will be put on sale
+const fractionId = 13601086205829910384631083059047775411896320n;
+
+// Create the fractional sale order for a hypercert
```
[the remark this answers is not in the document]
Comment (Jipperism) [reply guides/createFractionalSaleMakerAsk.md:32]: Lowercase, see https://www.hypercerts.org/docs/what-are-hypercerts

File: guides/createFractionalSaleMakerAsk.md [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+// This will also take care of fetching and setting the order nonce correctly
+const order = await hypercertExchangeClient.createFractionalSaleMakerAsk({
+  startTime: Date.now() / 1000, // Use it to create an order that will be valid in the future (Optional, Default to now)
+  endTime: Date.now() / 1000 + 60 * 60, // If you use a timestamp in ms, the function will revert (this order will be valid for one hour)
+  price: '10', // Note that this is the price per hypercert unit in wei
```
[the remark this answers is not in the document]
Comment (Jipperism) [reply guides/createFractionalSaleMakerAsk.md:37]: Went through the docs, the closest I can find is something like https://www.hypercerts.org/docs/guide/sell#advanced-settings. Let me know what you think. Looping in @bitbeckers in case I missed it somewhere in the docs.

File: src/HypercertExchangeClient.ts (hunk 4 above)
Comment (Jipperism) [reply src/HypercertExchangeClient.ts:322]: I had a more in-depth look, and I originally misinterpreted what the `merkleTree` was referring to. It seems to only refer to the proofs that are obtained when using `signMultipleMakerOrders()` to create the orders. See https://docs.looksrare.org/developers/sdk/signMultipleMakerOrders for a more thorough explanation.

I thought we would use this for allowlist orders, but they are implemented using a strategy and the merkle tree proofs for the allowlist are provided using the `additionalParams` argument. So it looks like we will only need this in the foreseeable future, if we plan on supporting signing multiple sales (and buys) at once from the frontend. That would also include implementing the endpoint for storing the tree and retrieving proofs (see https://docs.looksrare.org/developers/sdk/signMultipleMakerOrders again).

We could decide to remove the functionality, and that means we can get rid of all `merkleTree` parameters in `HypercertExchangeClient`. It would be a breaking change however, as the function signatures would change, but I think it might be worth it.

Looping in @bitbeckers and @holkexyz to weigh in on whether we are going to need this functionality in the future.

**Comment by Jipperism** [comment]
@pheuberger Had a look at all the comments and implemented the feedback. Please close all conversations on the PR that you consider resolved.

> Looks like you're not only removing old code, but also bumping the version number and changing some magic number. I would advocate for even more granular commits, because I don't know why you're doing any of that (except the version bump).

I updated the git commit description for https://github.com/hypercerts-org/marketplace-sdk/pull/25/commits/fe6ed4b8b680518fc64c081ed87f8ae3f39f92c4, hopefully that makes things clearer. I think splitting this up in more granular commits would also entail purposely releasing broken versions to NPM, as each commit would require a version bump (otherwise we have version on github that don't exist on NPM)? I can see both sides of the story but maybe that's not necessary for now, let me know what you think.

File: README.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (pheuberger) [reply README.md:57]: Not sure to be honest. I think it's fine the way it is 👍

File: README.md [hunk 15]
```diff
@@ -1,101 +1,137 @@
...
+- `build` - Build the SDK
+- `test` - Run the tests
+- `docs` - Generate the documentation into the `doc` folder
+- `supabase:types:hypercerts` - Generate types for the `data-staging` database
+
+## Architecture
+
+### Lifecycle of an order
+
+1. User A creates maker order and signs it
```
Comment (pheuberger) [inline README.md:46]: All the numbered bullets are the same (1.)

File: guides/cancelOrders.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (pheuberger) [reply guides/cancelOrders.md:10]: Got it, thank you!

File: README.md (hunk 15 above)
Comment (Jipperism) [reply README.md:46]: Yep, but markdown renders that as a numbered list correctly. This way it's easier to add something in the middle if necessary.

Example: (I think you can click edit on this comment to see the source).
1. a
1. b
1. c

File: guides/createDirectFractionsSaleMakerAsk.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (pheuberger) [reply guides/createDirectFractionsSaleMakerAsk.md]: I think it's better.  What do you think @bitbeckers?

File: guides/createFractionalSaleMakerAsk.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (pheuberger) [reply guides/createFractionalSaleMakerAsk.md:37]: It's probably better than not having it. Perhaps it's not worth explaining it in more detail. I think what trips me up is that there are fractions of a hypercert and then you can also divide it up even further by selling/buying fractions (a number of units) out of that fraction.

File: typedoc.json [hunk 16]
```diff
@@ -1,13 +1,12 @@
...
   "out": "./doc",
   "githubPages": false,
   "readme": "none",
   "gitRemote": "https://github.com/hypercerts-org/marketplace-sdk",
   "disableSources": true,
   "cleanOutputDir": false,
 
-  "hideBreadcrumbs": true,
-  "hideInPageTOC": true
+  "projectDocuments": ["./guides/readme.md"],
```
Comment (pheuberger) [inline typedoc.json:11]: This one should also be all caps like above, shouldn't it?

File: README.md (hunk 15 above)
Comment (pheuberger) [reply README.md:46]: Ah, that's great, didn't know!

File: README.md [hunk 17]
```diff
@@ -1,101 +1,138 @@
...
+1. Order gets stored in data postgres DB
+1. Order will live in DB until deleted
+1. Order will be visible to other users as long as it's valid.
+1. An order being executed or canceled (or many other reasons) might render it being invalid.
+    1. User B fetches order from API
+    1. User B creates taker order for maker order
+    1. User B signs taker order against maker order using the `HypercertExchangeClient`
+    1. User B calls the `executeOrder` method on the `HypercertExchangeClient` with the taker order, which calls the `HypercertExchange` contract in turn.
+    1. The `HypercertExchange` contract verifies the signature and executes the order
+    1. The `HypercertExchange` contract will emit an event that the order has been executed, which is picked up by the indexer which revalidates the order and updates the errors codes and validity in the DB.
```
[the remark this answers is not in the document]
Comment (pheuberger) [reply README.md]: It looks like there's something missing in the changed sentence @Jipperism, or am I missing something?

`If the transaction rendered the order invalid for any further sales, the error codes and validity in the DB.`

File: guides/faq.md [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+
+Call the public api endpoint [/marketplace/order-nonce](https://api.hypercerts.org/spec/#/Marketplace/UpdateOrderNonce), and use this nonce directly.
+
+### ❓ What to do when the order is created and signed ?
+
+Use the public api endpoint [/marketplace/orders](https://api.hypercerts.org/spec/#/Marketplace/StoreOrder) to push the order to the database. After that, the order will be visible by everyone using the API. There is also a `hypercertExchangeClient.registerOrder()` utility method available.
+
+### ❓ Why do I need to call grantTransferManagerApproval ?
+
+When you approve a collection to be traded on the hypercerts exchange, you approve the TransferManager instead of the exchange. Calling `grantTransferManagerApproval` gives the exchange contract the right to call the transfer function on the TransferManager. You need to call this function only once, the first time you use the V2.
```
[the remark this answers is not in the document]
Comment (pheuberger) [reply guides/faq.md]: It looks like you removed V2. Was that a leftover from LR?

File: src/HypercertExchangeClient.ts (hunk 4 above)
Comment (pheuberger) [reply src/HypercertExchangeClient.ts:322]: Thank you for the thorough explanation! Also waiting for @bitbeckers and @holkexyz to chime in here.

**Comment by pheuberger** [comment]
> @pheuberger Had a look at all the comments and implemented the feedback. Please close all conversations on the PR that you consider resolved.
> 
> > Looks like you're not only removing old code, but also bumping the version number and changing some magic number. I would advocate for even more granular commits, because I don't know why you're doing any of that (except the version bump).
> 
> I updated the git commit description for [fe6ed4b](https://github.com/hypercerts-org/marketplace-sdk/commit/fe6ed4b8b680518fc64c081ed87f8ae3f39f92c4), hopefully that makes things clearer. I think splitting this up in more granular commits would also entail purposely releasing broken versions to NPM, as each commit would require a version bump (otherwise we have version on github that don't exist on NPM)? I can see both sides of the story but maybe that's not necessary for now, let me know what you think.

Thank you, @Jipperism! Can you explain why we would need to release broken versions to NPM? Not every commit needs to be a release, nor have a version bump, unless I'm missing something?

File: README.md [hunk 19]
```diff
@@ -1,101 +1,137 @@
...
 
-```ts
-import { Addresses } from "@looksrare/sdk-v2";
-const addresses: Addresses = {...};
-const looksrare = new LooksRare(ChainId.HARDHAT, provider, signer, addresses);
+```bash
+cd marketplace-sdk
+pnpm install && pnpm run build
+cd ../your-project
+pnpm link @hypercerts-org/marketplace-sdk
```
Comment (pheuberger) [inline README.md:31]: This is actually wrong. It needs to say 

```bash
pnpm link <path to marketplace sdk on you local filesystem>
```

File: src/HypercertExchangeClient.ts (hunk 4 above)
Comment (bitbeckers) [reply src/HypercertExchangeClient.ts:322]: I wouldn't invest time in these kinda of refactors. Is we're going to invest significant amounts of time in the marketplace SDK it would be migrate it to `viem` in the regular SDK.

File: LICENSE.md
Comment (bitbeckers) [inline LICENSE.md:1]: License should also acknowledge Looksrare source repo

File: README.md [hunk 20]
```diff
@@ -1,101 +1,137 @@
...
-## Install
 
-This package has a peer dependency on [etherjs V5](https://docs.ethers.io/v5/).
+## SDK usage
 
-```bash
-yarn add @looksrare/sdk-v2 ethers
-```
+<!-- TODO: Check if links still work on deployment -->
+Read the [guides](./guides) if you need help with the implementation.
```
Comment (bitbeckers) [inline README.md:11]: These will reference to the hypercerts.org/docs, right?

File: README.md [hunk 21]
```diff
@@ -1,101 +1,137 @@
...
-
-// Create LooksRare instance using this provider
-const looksrare = new LooksRare(ChainId.HARDHAT, provider, signer, addresses);
+### Scripts
+
+- `dev` - Run the SDK in development mode
+- `build` - Build the SDK
+- `test` - Run the tests
+- `docs` - Generate the documentation into the `doc` folder
+- `supabase:types:hypercerts` - Generate types for the `data-staging` database
```
Comment (bitbeckers) [inline README.md:40]: Due to my late review these are now outdated? Because we connect to the API, not Supabase?

File: README.md [hunk 22]
```diff
@@ -1,101 +1,137 @@
...
-
-## Resources
-
-🔗 [Developer documentation](https://docs.looksrare.org/developers/welcome)
-
-🔗 [Public API documentation](https://looksrare.dev/reference/important-information)
+### Data
 
-🔗 [Developer discord](https://discord.gg/jJA4qM5dXz)
+Order information and nonces live in the `data-staging` and `data-production` database. Nonces are invalidated on-chain, but keeping track of the current nonce for a user happens off-chain in the DB. Orders do not live on-chain, but as they are signed they can be verified.
```
Comment (bitbeckers) [inline README.md:137]: Don't think this is relevant for a readme. If somebody forks this and wants to use a differen DB setup that's fine

File: guides/cancelOrders.md [hunk 23]
```diff
@@ -49,25 +32,21 @@ const receipt = await tx.wait();
...
+const tx = await hypercertExchangeClient.cancelAllOrders(true, true).call();
 const receipt = await tx.wait();
 ```
 
-For more details on our triple-nonce system, see the related documentation: [Triple-nonce system](https://docs.looksrare.org/developers/protocol/triple-nonce-system-v2)
-
-## Need help?
-
-You can reach out to the LooksRare team via our Developers Discord: [https://discord.gg/LooksRareDevelopers](https://discord.gg/LooksRareDevelopers)
+For more details on the triple-nonce system, see the related documentation: [Triple-nonce system](https://docs.looksrare.org/developers/protocol/triple-nonce-system-v2)
```
Comment (bitbeckers) [inline guides/cancelOrders.md:52]: Let's host that page ourselves

File: guides/createFractionalSaleMakerAsk.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (bitbeckers) [reply guides/createFractionalSaleMakerAsk.md:37]: There's also documentation on this on the specific strategy contract. It might be good to explain here what the strategy does

File: guides/executeTrade.md [hunk 24]
```diff
@@ -1,26 +1,58 @@
...
+---
+
+> [!WARNING]
+>
+> These code snippets are just examples and the data should never be used as is
 
 # How to create a Taker order and execute a trade
 
-Trades are executed on-chain by matching a `Maker` order with a `Taker` order. The maker order can be retrieved from the API, see [get v2 orders](https://looksrare.dev/v2/reference/getorders) for more details. While the taker order can be obtained by calling the `createTaker` method as shown here:
+Trades are executed on-chain by matching a `Maker` order with a `Taker` order. The maker order can be retrieved from the [graphQL API](https://staging-api.hypercerts.org/v1/graphql). The order used here has the same structure as a [maker order](../src/types.ts#L116), but with an added field `signature: string`. While the taker order can be obtained by calling the `createTaker` method as shown here:
```
Comment (bitbeckers) [inline guides/executeTrade.md:12]: Link to production API, or both. But staging could always be broken and not used as reference

File: src/HypercertExchangeClient.ts [hunk 25]
```diff
@@ -209,7 +201,7 @@ export class HypercertExchangeClient {
       endTime: endTime,
       price: price,
       itemIds: itemIds,
-      amounts: amounts,
+      amounts: itemIds.map(_ => 1n),
```
Comment (bitbeckers) [inline src/HypercertExchangeClient.ts:204]: Does this make sense? is it always 1? Will this break when we enable other strategies?

File: src/HypercertExchangeClient.ts (hunk 4 above)
Comment (bitbeckers) [reply src/HypercertExchangeClient.ts:322]: Reading this again, we also have merkle proofs in strategies to allow for users to determin to whom they want to sell.

File: src/utils/api.ts [hunk 26]
```diff
@@ -4,6 +4,7 @@ import { Database as HypercertsDatabase } from "./hypercerts-database-types";
 import { CONSTANTS, parseClaimOrFractionId } from "@hypercerts-org/sdk";
 import { getFractionsById, getOrders } from "./graphl";
 import { cacheExchange, Client, fetchExchange } from "@urql/core";
+import { deprecate } from "util";
```
Comment (bitbeckers) [inline src/utils/api.ts:7]: unused import? Type from @deprecated?

File: README.md (hunk 21 above)
Comment (Jipperism) [reply README.md:40]: Yes and no. We still use the generated types in 1 or 2 places, but never connect to supabase anymore once https://github.com/hypercerts-org/marketplace-sdk/pull/27 gets merged. It could be removed if we pull out the 1 or 2 types that we're still using but I think that belongs in the other PR.

File: src/utils/api.ts (hunk 26 above)
Comment (Jipperism) [reply src/utils/api.ts:7]: Good catch. Got imported automatically when trying to mark a method as deprecated in the js docs.

File: README.md (hunk 19 above)
Comment (Jipperism) [reply README.md]: Addressed

File: LICENSE.md
Comment (Jipperism) [reply LICENSE.md:1]: Addressed

File: README.md (hunk 20 above)
Comment (Jipperism) [reply README.md:11]: Yes, will update this when moving the guides to @hypercerts-org/docs.

File: README.md (hunk 22 above)
Comment (Jipperism) [reply README.md]: Addressed

File: guides/executeTrade.md (hunk 24 above)
Comment (Jipperism) [reply guides/executeTrade.md]: Addressed

**Comment by Jipperism** [comment]
@bitbeckers GitHub UI is somehow not letting me respond to

>Reading this again, we also have merkle proofs in strategies to allow for users to determin to whom they want to sell.

The allowlist here refers to a different type. Namely, the one that gets created when using `signMultipleMakerOrders` from the SDK. This should only be used programatically according to the devs of looksrare.

So yes, in the future we might need a param `merkleTree`, but this is not referring to that specific one.

File: src/HypercertExchangeClient.ts (hunk 4 above)
Comment (Jipperism) [reply src/HypercertExchangeClient.ts:322]: The allowlist here refers to a different type. Namely, the one that gets created when using signMultipleMakerOrders from the SDK. This should only be used programatically according to the devs of looksrare.

So yes, in the future we might need a param merkleTree, but this is not referring to that specific one.

File: README.md (hunk 21 above)
Comment (bitbeckers) [reply README.md:40]: https://github.com/hypercerts-org/marketplace-sdk/issues/29

File: guides/createFractionalSaleMakerAsk.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (Jipperism) [reply guides/createFractionalSaleMakerAsk.md]: Addressed

File: guides/cancelOrders.md (hunk 23 above)
Comment (bitbeckers) [reply guides/cancelOrders.md:52]: Write our own? What if their page goes down?

File: guides/faq.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (Jipperism) [reply guides/faq.md]: Yes, it was.

File: typedoc.json (hunk 16 above)
Comment (Jipperism) [reply typedoc.json]: Addressed

File: src/HypercertExchangeClient.ts (hunk 25 above)
Comment (bitbeckers) [inline src/HypercertExchangeClient.ts:204]: Resolved the conversation, but this is an update in business logic. So create a unit test, release an alpha and validate on testnet.

File: src/utils/calls/nonces.ts [hunk 27]
```diff
@@ -30,23 +30,6 @@ export const cancelOrderNonces = (
   };
 };
 
-export const cancelSubsetNonces = (
```
Comment (bitbeckers) [inline src/utils/calls/nonces.ts:33]: Missed this last time, why was it removed?

File: guides/faq.md [hunk 28]
```diff
@@ -0,0 +1,24 @@
+---
+title: FAQ
+group: Guides
+---
+
+# FAQ
+
+### ❓ How to retrieve order nonce ?
+
+Call the public API endpoint [/marketplace/order-nonce](https://api.hypercerts.org/spec/#/Marketplace/UpdateOrderNonce), and use this nonce directly.
```
Comment (bitbeckers) [inline guides/faq.md:10]: /v1/

File: guides/faq.md [hunk 29]
```diff
@@ -0,0 +1,24 @@
...
+
+### ❓ What to do when the order is created and signed ?
+
+Use the public API endpoint [/marketplace/orders](https://api.hypercerts.org/spec/#/Marketplace/StoreOrder) to push the order to the database. After that, the order will be visible by everyone using the API. There is also a `hypercertExchangeClient.registerOrder()` utility method available.
```
Comment (bitbeckers) [inline guides/faq.md:14]: /v1/

File: guides/faq.md [hunk 30]
```diff
@@ -0,0 +1,24 @@
...
+
+### ❓ Why do I need to call grantTransferManagerApproval ?
+
+When you approve a collection to be traded on the hypercerts exchange, you approve the TransferManager instead of the exchange. Calling `grantTransferManagerApproval` gives the exchange contract the right to call the transfer function on the TransferManager. You need to call this function only once, the first time you use the Hypercerts Exchange Protocol.
+
+## Resources
+
+🔗 [Public API documentation](https://api.hypercerts.org/spec)
+
+🔗 [Developer discord](https://discord.gg/PZtJ6W9S)
```
Comment (bitbeckers) [inline guides/faq.md:24]: Invite invalid

File: guides/createMakerAsk.md [hunk 31]
```diff
@@ -1,57 +1,71 @@
...
 
-const lr = new LooksRare(ChainId.MAINNET, provider, signer);
+// The ID of the fraction to be sold
+const fractionId = 13601086205829910384631083059047775411896320n;
 
-const { maker, isCollectionApproved, isTransferManagerApproved } = await lr.createMakerAsk({
-  collection: "0x0000000000000000000000000000000000000000", // Collection address
-  collectionType: CollectionType.ERC721,
+const { maker, isCollectionApproved, isTransferManagerApproved } = await hypercertExchangeClient.createMakerAsk({
+  collection: "0x123", // Collection address
```
Comment (bitbeckers) [inline guides/createMakerAsk.md:39]: Same pet peeve as in other docs, actually use addresses because everytime I read this I think the example is incorrect

File: guides/createFractionalSaleMakerAsk.md [hunk 32]
```diff
@@ -0,0 +1,122 @@
...
+
+const hypercertExchangeClient = new HypercertExchangeClient(ChainId.OPTIMISM, provider, signer);
+
+// The ID of the fraction that will be put on sale
+const fractionId = 13601086205829910384631083059047775411896320n;
+
+// Create the fractional sale order for a hypercert
+// This will also take care of fetching and setting the order nonce correctly
+const order = await hypercertExchangeClient.createFractionalSaleMakerAsk({
+  startTime: Date.now() / 1000, // Use it to create an order that will be valid in the future (Optional, Default to now)
```
Comment (bitbeckers) [inline guides/createFractionalSaleMakerAsk.md:37]: We discussed yesterday that that would fail right? Taking blocktimes and on chain validations into account?

File: guides/createDirectFractionsSaleMakerAsk.md [hunk 33]
```diff
@@ -0,0 +1,111 @@
...
+// The ID of the fraction that will be put on sale
+const fractionId = 13601086205829910384631083059047775411896320n;
+
+// Create the fractional sale order for a hypercert
+// This will also take care of fetching and setting the order nonce
+const order = await hypercertExchangeClient.createDirectFractionsSaleMakerAsk({
+  startTime: Date.now() / 1000, // Use it to create an order that will be valid in the future (Optional, defaults to the current timestamp in seconds)
+  endTime: Date.now() / 1000 + 60 * 60, // If you use a timestamp in ms, the function will revert (this order will be valid for one hour)
+  itemIds: [fractionId], // ID of the fraction that will be put on sale
+  price: parseUnits("1", 18), // Price for all units in the hypercert fraction, in this example we're selling for 1 USDC.
```
Comment (bitbeckers) [inline guides/createDirectFractionsSaleMakerAsk.md:38]: actually wrong decimals for USDC

**Review by bitbeckers — CHANGES_REQUESTED** [review]

File: src/utils/calls/nonces.ts (hunk 27 above)
Comment (Jipperism) [reply src/utils/calls/nonces.ts:33]: Because we don't support subset nonces in our api.

File: guides/faq.md (hunk 28 above)
Comment (Jipperism) [reply guides/faq.md]: Fixed

File: guides/faq.md (hunk 29 above)
Comment (Jipperism) [reply guides/faq.md]: Fixed

File: guides/faq.md (hunk 30 above)
Comment (Jipperism) [reply guides/faq.md]: Fixed, created a non-expiring invite.

File: guides/createFractionalSaleMakerAsk.md (hunk 32 above)
Comment (Jipperism) [reply guides/createFractionalSaleMakerAsk.md]: Fixed, made it 5 minutes into the future now.

File: guides/createDirectFractionsSaleMakerAsk.md (hunk 33 above)
Comment (Jipperism) [reply guides/createDirectFractionsSaleMakerAsk.md]: Fixed. Impressed with the catch.

File: README.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (Jipperism) [reply README.md]: For example: fraction doesn't have enough units left in it to do another sale. Also, fixed the missing part in the sentence.

File: guides/createFractionalSaleMakerAsk.md [hunk 34]
```diff
@@ -0,0 +1,120 @@
...
+  currency: ZeroAddress, // Currency address (0x0 for ETH)
+});
+
+// Grant the TransferManager the right the transfer assets on behalf of the Hypercert Exchange Protocol
+if (!isTransferManagerApproved) {
+  const tx = await hypercertExchangeClient.grantTransferManagerApproval().call();
+  await tx.wait();
+}
+
+// Approve the collection items to be transferred by the TransferManager
```
[the remark this answers is not in the document]
Comment (Jipperism) [reply guides/createFractionalSaleMakerAsk.md]: A collection in this context is not the same as our own collections. In the case of hypercerts, they all belong to the collection [0xa16DFb32Eb140a6f3F2AC68f41dAd8c7e83C4941](https://sepolia.etherscan.io/address/0xa16DFb32Eb140a6f3F2AC68f41dAd8c7e83C4941) (hypercert-minter-address). The original usecase, it would probably refer to the NFTs minter address.

File: guides/cancelOrders.md (hunk 23 above)
Comment (Jipperism) [reply guides/cancelOrders.md]: Added the docs.

**Comment by bitbeckers** [comment]
As you've release 0.4.0 that'd also need to be merged in here since this PR also influences business logic

File: src/HypercertExchangeClient.ts (hunk 25 above)
Comment (Jipperism) [reply src/HypercertExchangeClient.ts:204]: Tested and works in https://github.com/hypercerts-org/hypercerts-app/pull/428.

Comment (bitbeckers) [reply src/HypercertExchangeClient.ts:204]: Wait they're completely broken? Let's create an issue for that, we should be able to get green tests and automate package updates from our CI

Comment (Jipperism) [reply src/HypercertExchangeClient.ts:204]: Yes, they have been almost since the beginning, and we've spent some time trying to get it to work as well but that was forever ago. Biggest issue is hardhat setup within the tests.

Comment (Jipperism) [reply src/HypercertExchangeClient.ts:204]: Created an issue here: https://github.com/hypercerts-org/marketplace-sdk/issues/30

File: guides/cancelOrders.md (hunk 23 above)
Comment (Jipperism) [reply guides/cancelOrders.md]: @bitbeckers added it back after discussion with @holkexyz . Kept the missing subset nonce docs in there as well commented out, so we can show that once we've implemented the API part of it.

**Comment by Jipperism** [comment]
@bitbeckers has been updated to incorporate changes from 0.4.1
